app/src/main/java/com/jvoye/tasky/agenda_detail/presentation/AgendaDetailScreenViewModel.kt

```diff
@@ -31,64 +29,60 @@ class AgendaDetailScreenViewModel(
     private val savedStateHandle: SavedStateHandle
 ): ViewModel() {
     private val _state = MutableStateFlow(AgendaDetailState(
-
+        titleText = savedStateHandle["titleText"],
+        descriptionText = savedStateHandle["descriptionText"],
+        isEditMode = savedStateHandle.get<Boolean>("isEditMode") ?: false,
+        selectedDateMillis = savedStateHandle.get<Long>("selectedDateMillis"),
+        notificationType = savedStateHandle.get<NotificationType>("notificationType") ?: NotificationType.THIRTY_MINUTES_BEFORE
     ))
     private var hasLoadedInitialData = false
 
     val state = _state
         .onStart {
             if (!hasLoadedInitialData) {
-                // Try to restore TaskyItem from SavedStateHandle first
-                val restoredTaskyItem = savedStateHandle.get<TaskyItem>(TASKY_ITEM_KEY)
-
-                if (restoredTaskyItem != null) {
+                val restoredItemId = savedStateHandle.get<Long>("itemId")
+                if (restoredItemId != null) {
                     _state.update { it.copy(
-                        taskyItem = restoredTaskyItem
+                         taskyItem = it.taskyItem.copy(
+                             id = restoredItemId
+                         )
                     )}
                 } else {
-                    // If not available in SavedStateHandle, loading from repository
+                    // If not available in SavedStateHandle, loading data from repository
                     getTaskyItemFromRepo(taskyItemId)
+                    savedStateHandle["itemId"] = taskyItemId
+
                 }
+
                 getTypeAndEditMode(isEdit, taskyType)
                 hasLoadedInitialData = true
             }
         }
-        .onEach {
-            savedStateHandle[TASKY_ITEM_KEY] = it.taskyItem
-            println("TASKY ITEM SAVED STATE: ${it.taskyItem}")
+        .onEach { state ->
+            savedStateHandle["titleText"] = state.titleText
```

Comment:
I’d suggest moving those string keys into a private companion object. That way you centralize them in one place 😊
```kotlin
private companion object {
    const val KEY_TITLE = "titleText"
    const val KEY_DESC = "descriptionText"
    const val KEY_DATE = "selectedDateMillis"
    const val KEY_NOTIFY = "notificationType"
    const val KEY_ITEM_ID = "itemId"
}
```